src/Workspaces/CSharp/Portable/Extensions/SimpleNameSyntaxExtensions.cs

```diff
@@ -15,7 +15,15 @@ public static ExpressionSyntax GetLeftSideOfDot(this SimpleNameSyntax name)
             Contract.Requires(name.IsMemberAccessExpressionName() || name.IsRightSideOfQualifiedName() || name.IsParentKind(SyntaxKind.NameMemberCref));
             if (name.IsMemberAccessExpressionName())
             {
-                return ((MemberAccessExpressionSyntax)name.Parent).Expression;
+                var conditionalAccess = name.Parent.Parent as ConditionalAccessExpressionSyntax;
```

Comment:
Is this only an issue if the `ConditionalAccessExpressionSyntax` is the grandparent?  What if it's further up a chain of dotting?